### PR TITLE
Postjob

### PR DESCRIPTION
**Pull request by majakudlicka**
Saving data from PostJob form into the database

### CODE REVIEW

File: database/database_queries.js [hunk 1]
```diff
@@ -37,15 +37,17 @@ data.postStudents = (object, callback) => {
...
-        [object.last_name],
-        [object.email],
-        [object.dob],
-        [object.uni],
-        [object.job_cat],
-        [object.picture],
-        [object.bio],
-        [object.phone],
+        [
+            object.first_name,
```
Comment (des-des) [inline database/database_queries.js]: object is bad name

File: database/database_queries.js [hunk 2]
```diff
@@ -59,14 +61,16 @@ data.postResidents = (object, callback) => {
...
-        [object.first_name],
-        [object.last_name],
-        [object.email],
-        [object.address],
-        [object.dob],
-        [object.picture],
-        [object.bio],
-        [object.phone],
+        [
+            object.first_name,
```
Comment (des-des) [inline database/database_queries.js]: object is bad name

File: database/database_queries.js [hunk 3]
```diff
@@ -78,19 +82,24 @@ data.postResidents = (object, callback) => {
...
-        [object.start_time],
-        [object.end_date],
-        [object.end_time],
-        [object.description],
-        [object.rate],
-        [object.resident_id],
+        `INSERT INTO jobs(job_title, start_date, start_time, end_date, end_time, description,
+                    rate, resident_id, category) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)`,
+        [
+            object.job_title,
```
Comment (des-des) [inline database/database_queries.js]: object is bad name

File: src/client/actions/post_job.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+import {POST_JOB} from '../constants/action_types.js';
+
+export const postJob = job => dispatch => {
+    console.log('iside PostJOb');
```
Comment (des-des) [inline src/client/actions/post_job.js]: :fire:

File: src/client/components/Form_Post_Job.js [hunk 5]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        var jobData = this.state.jobData;
+        jobData.start_date = evt.target.value;
+        this.setState({jobData});
+        this.props.history.push('/');
+    }
+
+    _onChangeStartTime(evt) {
+        var jobData = this.state.jobData;
+        jobData.start_time = evt.target.value;
+        this.setState({jobData});
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: You should not mutate the state object
```js
this.setState({
  jobData: {
     ...jobData,
     start_time: evt.target.value
  }
```
or

```js
this.setState({
  jobData: Object.assign({}, jobData, { start_time: evt.target.value })
})
```

File: src/client/components/Form_Post_Job.js [hunk 6]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+    }
+
+    _onChangeEndDate(evt) {
+        var jobData = this.state.jobData;
+        jobData.end_date = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same as above

File: src/client/components/Form_Post_Job.js [hunk 7]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        this.setState({jobData});
+    }
+
+    _onChangeEndTime(evt) {
+        var jobData = this.state.jobData;
+        jobData.end_time = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same as above

File: src/client/components/Form_Post_Job.js [hunk 8]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        this.setState({jobData});
+    }
+
+    _onChangeTitle(evt) {
+        var jobData = this.state.jobData;
+        jobData.job_title = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same as above

File: src/client/components/Form_Post_Job.js [hunk 9]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        this.setState({jobData});
+    }
+
+    _onChangeDescription(evt) {
+        var jobData = this.state.jobData;
+        jobData.description = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same

File: src/client/components/Form_Post_Job.js [hunk 10]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        this.setState({jobData});
+    }
+
+    _onChangeRate(evt) {
+        var jobData = this.state.jobData;
+        jobData.rate = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same

File: src/client/components/Form_Post_Job.js [hunk 11]
```diff
@@ -2,11 +2,92 @@ import React, {Component} from 'react';
...
+        this.setState({jobData});
+    }
+
+    _onChangeCategory(evt) {
+        var jobData = this.state.jobData;
+        jobData.category = evt.target.value;
```
Comment (des-des) [inline src/client/components/Form_Post_Job.js]: same

File: src/client/components/pages/PostJob.js [hunk 12]
```diff
@@ -1,13 +1,25 @@
...
+//     rate: '100',
+//     resident_id: '1',
+//     category: 'photography'
+// };
+
 class PostJob extends Component {
     render() {
-        const dispatch = this.props.dispatch;
-        const {formState, currentlySending} = this.props.data;
+        // const dispatch = this.props.dispatch;
```
Comment (des-des) [inline src/client/components/pages/PostJob.js]: :fire:

File: src/client/reducers/home.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+ *       stateVariable: action.var
+ *   });
+ */
+
+import {
+    CHANGE_FORM,
+    SET_AUTH,
+    SENDING_REQUEST,
+    SET_ERROR_MESSAGE
+} from '../constants/AppConstants';
```
Comment (des-des) [inline src/client/reducers/home.js]: Put this in `../constants/action_types.js`

File: src/client/reducers/home.js [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    currentlySending: false,
+    loggedIn: beevrAPI.loggedIn(),
+    errorMessage: ''
+};
+
+// Takes care of changing the application state
+export default function homeReducer(state = initialState, action) {
+    switch (action.type) {
+    case CHANGE_FORM:
+        return assign({}, state, {
```
Comment (des-des) [inline src/client/reducers/home.js]: You need to choose a style

Comment (des-des) [inline src/client/reducers/home.js]:
```js
{ ...state, { k: 'v' } }
```
or
```js
Object.assign({}, state, {k: 'v'})
```

File: src/server/server.js [hunk 15]
```diff
@@ -6,34 +6,23 @@ const inert = require('inert');
...
 server.connection({
     port: PORT
     /*tls: process.env.NODE_ENV !== 'production' && {
-        key: fs.readFileSync('./keys/key.pem'),
-        cert: fs.readFileSync('./keys/cert.pem'),
-    },
-    state: {
-    isSameSite: 'Lax',
-    },*/
+      key: fs.readFileSync('./keys/key.pem'),
```
Comment (des-des) [inline src/server/server.js]: delete this

File: src/server/server.js [hunk 16]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
-                }
-                else reply({
+            data.getJobs(request.url.query.term, (err, res) => {
+                if (err)
+                    reply.status(500)(
+                        'Failed to connect load data from the database'
+                    );
+                console.log(err);
+                // else {
+                reply({
```
Comment (des-des) [inline src/server/server.js]: Do not reply twice!

File: src/server/server.js [hunk 17]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+                // }
+            });
+        }
+    });
+
+    server.route({
+        method: 'POST',
+        path: '/api/jobs',
+        handler: (request, reply) => {
+            console.log(request.payload);
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: src/server/server.js [hunk 18]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+            data.postJobs(request.payload, (err, res) => {
+                if (err) console.log(error);
+                // reply.status(500)(
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: src/server/server.js [hunk 19]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+                //     'Failed to connect load data from the database'
+                // );
+                console.log(err);
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: src/server/server.js [hunk 20]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+                // else {
+                console.log(res);
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: src/server/server.js [hunk 21]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+        path: '/api/jobs',
+        handler: (request, reply) => {
+            console.log(request.payload);
+            data.postJobs(request.payload, (err, res) => {
+                if (err) console.log(error);
+                // reply.status(500)(
+                //     'Failed to connect load data from the database'
+                // );
+                console.log(err);
+                // else {
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: src/server/server.js [hunk 22]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+                console.log(res);
+                reply({
```
Comment (des-des) [inline src/server/server.js]: Do not reply twice.

This is wrong.

The if / else branching was correct

File: src/server/server.js [hunk 23]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
+    server.route({
+        method: 'GET',
+        path: '/api/random_jobs',
+        handler: (request, reply) => {
+            data.getRandomJobs((err, res) => {
+                if (err)
+                    reply.status(500)(
+                        'Failed to connect load data from the database'
+                    );
+                else {
```
Comment (des-des) [inline src/server/server.js]: Yes, this one is correct

File: src/server/server.js [hunk 24]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
-                    reply(err.message);
-                }
-                else reply({
+            data.getJobs(request.url.query.term, (err, res) => {
+                if (err)
+                    reply.status(500)(
+                        'Failed to connect load data from the database'
+                    );
+                console.log(err);
+                // else {
```
Comment (des-des) [inline src/server/server.js]: ??

File: src/server/server.js [hunk 25]
```diff
@@ -70,17 +59,62 @@ server.register(plugins, err => {
...
-                        Aborting`);
-                    reply(err.message);
-                }
-                else reply({
+            data.getJobs(request.url.query.term, (err, res) => {
+                if (err)
+                    reply.status(500)(
+                        'Failed to connect load data from the database'
+                    );
+                console.log(err);
```
Comment (des-des) [inline src/server/server.js]: :fire:

File: database/database_queries.js (hunk 1 above)
Comment (majakudlicka) [inline database/database_queries.js]: Renamed

File: database/database_queries.js (hunk 2 above)
Comment (majakudlicka) [inline database/database_queries.js]: Renamed

File: database/database_queries.js (hunk 3 above)
Comment (majakudlicka) [inline database/database_queries.js]: Renamed

File: src/client/components/Form_Post_Job.js (hunk 5 above)
Comment (majakudlicka) [inline src/client/components/Form_Post_Job.js]: Thanks, very helpful!!

File: src/client/reducers/home.js (hunk 13 above)
Comment (majakudlicka) [inline src/client/reducers/home.js]: Done

File: src/client/reducers/home.js (hunk 14 above)
Comment (majakudlicka) [inline src/client/reducers/home.js]: Done

File: src/server/server.js (hunk 15 above)
Comment (majakudlicka) [inline src/server/server.js]: Done

File: src/server/server.js (hunk 22 above)
Comment (majakudlicka) [inline src/server/server.js]: Ok

File: src/server/server.js (hunk 23 above)
Comment (majakudlicka) [inline src/server/server.js]: Ok copied that logic to other routes

**Comment by des-des** [comment]
no changes

**Comment by des-des** [comment]
no changes

**Pull request merged**